bigtable-client-core-parent/bigtable-hbase/src/test/java/com/google/cloud/bigtable/hbase/wrappers/veneer/TestBulkReadVeneerApi.java

```diff
@@ -76,20 +76,17 @@ public void tearDown() throws InterruptedException {
   public void testAdd() throws Exception {
     BulkReadVeneerApi bulkReadWrapper =
         new BulkReadVeneerApi(BigtableDataClient.create(settingsBuilder.build()), TABLE_ID);
-    ApiFuture<Result> resultFuture1 = bulkReadWrapper.add(ByteString.copyFromUtf8("one"), null);
-    assertFalse(resultFuture1.isDone());
+    ApiFuture<Result> resultFuture = bulkReadWrapper.add(ByteString.copyFromUtf8("one"), null);
+    assertFalse(resultFuture.isDone());
+    // Here AutoFlush triggers the batch
+    resultFuture.get();
 
     Filters.Filter filter = Filters.FILTERS.key().regex("cf");
-    ApiFuture<Result> secBatchResult1 = bulkReadWrapper.add(ByteString.copyFromUtf8("1"), filter);
-    ApiFuture<Result> secBatchResult2 = bulkReadWrapper.add(ByteString.copyFromUtf8("2"), filter);
```

Comment:
how come we are removing the second batch?

Reply:
I think this test currently flawed. There is a race condition between the response setting the 2nd item's future and us checking it. 

I think this entire thing can be simplified by calling flush instead of relying on the autoflush behavior